hyperspy/_signals/signal1d.py

```diff
@@ -1532,6 +1533,63 @@ def estimating_function(spectrum,
 
     estimate_peak_width.__doc__ %= (SHOW_PROGRESSBAR_ARG, NUM_WORKERS_ARG)
 
+    def get_intensity(self,
+                           indexes,
+                           start=None,
+                           stop=1.0,
+                           factor=1.0,
+                           norm=None,
+                           minimum_intensity=None,
+                           **kwargs):
+        """
+        A generic function for getting the intensity of a line at different positions in the Signal1D.
+
+        This function is used mostly for plotting markers and the output can be directly
+        passed to a `MarkerCollection` object using the `MarkerCollection.from_signal()` method.
+
+        Parameters
+        ----------
+        indexes : list of float
+            The indexes of the positions of the line in the signal..
+        factor : float or list of float
+            The factor to multiply the intensity by. If a list is passed, it must have the same
+            length as `indexes`.
+        str, optional
+            Normalization to apply to the intensities. Can be 'log' or None.
+        minimum_intensity : float, optional
+            Minimum intensity to use when norm is 'log'.
+        start : float, optional
+            Start value for scaling the vertical line
+        stop : float, optional
+            Stop value for scaling the vertical line or height of the point.
+        real_index : 1D numpy array
+            The real values for the indexes in calibrated units.
```

Comment:
Are all these parameters necessary? As it is written at the moment, there are not clear, for example, what are the vertical line and the scaling.

Reply:
Let's just get rid of this now?  I found this pattern in a couple of different packages for finding the intensity at some points for labeling data.  This is mostly replaced by the ""relative" markers. 

That being said there are times you might want to find the intensity at some edge or excitation energy.  Then again I would rather that just be something like `s.isig[(1.5,2.5,4.2)]`

Reply:
Can we leave it for another time for the sake of finishing this PR?! 😉

Reply:
@ericpre  Yes :) I've deleted that portion and will push the new changes once I get all of the tests passing